### DISCUSSION TITLE
Add non-vue leaflet object to vue-leaflet component

### DISCUSSION THREAD

**Discussion by rr326** [vue-leaflet/vue-leaflet#383] [Q&A]
I am using vue-leaflet (3) with Vue 3 and the composition api.

I have a parent component that uses <l-map> and draws a map. Good.
I am trying to use [this](https://github.com/coyotesqrl/Leaflet.ArrowCircle?tab=readme-ov-file) leaflet plugin along with it. I create a maker and I think it works. The last step is: `  .addTo(mymap)`.

I can't figure out how to do this.

Most of the posts I've read online talk about using `this.$refs...` but in my composition API, it looks like `this` is undefined. I've tried to follow all the template refs docs, and I'm missing something. 

I would guess that what I am trying to do is probably quite easy, but I'm clearly missing something.

Thanks for any help.

**Comment by brentporter** [comment]
Have you tried to use the map objects addLayer() method as opposed to .addTo()?
so it would be
`mymap.addLayer(SomeLayer)`

**Comment by rr326** [comment]
@brentporter YES! Thank you so much. I was at wits end.

I'll clean up my code and put something up here in case anyone else has the same problem.

**Reply by baillysi** [reply]
@rr326 Could you share your code ? I would be interested, facing a similar problem. 
Thank's a lot ! Simon

**Comment by rr326** [comment]
@baillysi 

## Parent
```js
const map = ref(null)
```

```html
<l-map ref="map">
    <Child  :map="map" />
```

## Child

```js
const props = defineProps({
    map: {
        // ref to the map component. access with map.leafletObject
        type: Object,
        required: true,
    },
})

childEl = L.marker(...)
props.map.leafletObject.addLayer(childEl)
```

**Comment by rr326** [comment]
@brentporter 

Is there a way to access the parent ref (in the composition api) without passing it as a prop? Or find it by `id`?

**Reply by brentporter** [reply]
@rr326 I am sorry I didn't see this. I haven't tried using this pattern. I usually deal with state (properties and such) from Pinia stores and watchers/storeToRefs. After using refs to the nth degree in earlier versions of vue I am trying hard to stick to stores. Sry.
